### ISSUE TITLE
Naming collisions between temps and knots aren't tested for

### ISSUE THREAD

**Issue by joethephish** [inkle/ink#327]
```
LIST someList = A, B

~ temp heldItems = (A) 
{LIST_COUNT(heldItems)}

=== function heldItems()
~ return (A)
```

Results in: `RUNTIME ERROR: 'test.ink' line 4: Cannot perform operation 'LIST_COUNT' on Int`, because it's getting the read count of `heldItems`.

**Comment by joethephish** [comment]
Fixed in 1788a2310779548e8fd36afbc8f411f524eb912c.

**Comment by joethephish** [comment]
And in this commit I added much stronger name collision checks in general: b419295ddbf6baa49a6523a3db098c145420c8ad.
